packages/sp/folders/types.ts

```diff
@@ -162,10 +163,10 @@ export class _Folder extends _SharePointQueryableInstance<IFolderInfo> {
     @tag("f.moveTo")
     public async moveTo(destUrl: string): Promise<void> {
 
-        const { ServerRelativeUrl: srcUrl, ["odata.id"]: absoluteUrl } = await this.select("ServerRelativeUrl")();
-        const webBaseUrl = extractWebUrl(absoluteUrl);
-        const hostUrl = webBaseUrl.replace("://", "___").split("/")[0].replace("___", "://");
-        await spPost(Folder(webBaseUrl, "/_api/SP.MoveCopyUtil.MoveFolder()"),
+        const { ServerRelativeUrl: srcUrl } = await this.select("ServerRelativeUrl")();
+        const web = await sp.web.select("Url, ServerRelativeUrl")();
```

Comment:
This won't work, unfortunately. `sp.web` takes the current page web or a baseUrl-provided web. However, `this` in the class instance is not necessarily chained from a current web, but any defined for a parent API instance.